src/controllers/admin/office.ts

```diff
@@ -76,8 +79,94 @@ export const GetAllAdminHandler = async (req: Request, res: Response) => {
       admins: currentAdmins
     });
   } catch (e) {
+    console.log(e)
     return res.status(500).json({
       message: "Failed to retrieve admin details"
     });
   }
 };
+
+export const EditAdminHandler = async (req: Request & { user: { email: string; role: 'Student' | 'Faculty' } }, res: Response) => {
+  try {
+    const validBody = EditAdminValidator.safeParse(req.body);
+    
+    if (!validBody.success) {
+      res.status(400).json({
+        message: "Invalid input data",
+      });
+      throw new CustomError("Invalid input data");
+    }
+
+    return await prismaClient.$transaction(async (tx) => {
+      const adminToEdit = await tx.admin.findFirst({
+        where: {
+          email: req.params.email
+        },
+      });
+
+      if (!adminToEdit) {
+        res.status(404).json({
+          message: "Admin not found",
+        });
+        throw new CustomError("Admin not found");
+      }
+
+      if (req.user.role === "Student" && req.user.email !== req.params.email) {
```

Comment:
This if-statement is unnecessary. We are not collecting role from the client.

Reply:
Should retain mail then?